projects/ziti-console-lib/src/lib/features/visualizer/network-visualizer/network-visualizer.component.ts

```diff
@@ -0,0 +1,1337 @@
+                this.router_policies,
+                this.service_router_policies,
+                this.uniqId
+            );
+
+            this.uniqId = treeObj.lastId;
+            this.networkGraph = d3.hierarchy(treeObj, function (d) {
+                return d.children;
+            });
+
+            this.appwanstreeNodes = [];
+            this.servicesTreeNodes = [];
+            this.endpointsTreeNodes = [];
+            this.edgeroutersTreeNodes = [];
+            this.erPoliciesTreeNodes = [];
+            this.serviceEdgeRouterPolicyTreeNodes = [];
+
+            this.appwanstreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[0],
+                this.appwanstreeNodes,
+                'Service Policy'
+            );
+            this.servicesTreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[1],
+                this.servicesTreeNodes,
+                'Service'
+            );
+            this.endpointsTreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[2],
+                this.endpointsTreeNodes,
+                'Identity'
+            );
+            this.edgeroutersTreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[3],
+                this.edgeroutersTreeNodes,
+                'Edge Router'
+            );
+            this.erPoliciesTreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[4],
+                this.erPoliciesTreeNodes,
+                'Router Policy'
+            );
+
+            this.serviceEdgeRouterPolicyTreeNodes = this.fillAutoCompleteTreeNodes(
+                this.networkGraph.children[5],
+                this.serviceEdgeRouterPolicyTreeNodes,
+                'Service Router Policy'
+            );
+
+            this.initTopoView();
+        } catch (err) {
+            console.log('error',err);
+        }
+
+        this.isLoading = false;
+  }
+    fillAutoCompleteTreeNodes(nodeObj, arr, typeName) {
+        const name = nodeObj.data.name;
+        // const id = nodeObj.data.id;
+        if (!name) {
+            return;
+        }
+        if (typeof nodeObj.data.type === 'undefined') {
+            //      //
+        } else {
+            if (arr.indexOf(name) < 0 && nodeObj.data.type === typeName) {
+                arr.push(name);
+            }
+        }
+
+        const childNodes = nodeObj.children ? nodeObj.children : nodeObj._children;
+        for (let i = 0; childNodes && i < childNodes.length; i++) {
+            this.fillAutoCompleteTreeNodes(childNodes[i], arr, typeName);
+        }
+
+        return arr;
+    }
+
+  initTopoView() {
+        this.autocompleteOptions = [];
+        // const duration = 750;
+        const margin = { top: 20, right: 90, bottom: 30, left: 0 };
+        const width = 1600 - margin.left - margin.right;
+        const height = 1600 - margin.top - margin.bottom;
```

Comment:
We might want to make these dimensions more dynamic instead of a fixed value of `1600`

Reply:
yes. agree